### PR TITLE
861 segresnet

### PR DESCRIPTION
**Pull request by yiheng-wang-nv**
Implements #861 .

### Description
Under default settings (except utilizing batch norm), the model gets mean dice 0.9377 after 342 epochs in Spleen Dataset.

### Status
**Ready**

### Types of changes

- [ ] Non-breaking change (fix or new feature that would not break existing functionality).
- [x] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [x] Integration tests passed locally by running `./runtests.sh --codeformat --coverage`.
- [ ] In-line docstrings updated.
- [x] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/networks/blocks/segresnet_block.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        return norm
+
+
+def get_conv_layer(
+    spatial_dims: int, in_channels: int, out_channels: int, kernel_size: int = 3, stride: int = 1, bias: bool = False
+):
+
+    conv_type: Type[Union[nn.Conv1d, nn.Conv2d, nn.Conv3d]] = Conv[Conv.CONV, spatial_dims]
+
+    conv = conv_type(
```
Comment (wyli) [inline monai/networks/blocks/segresnet_block.py:40]: could line 38-42 be implemented with `monai.networks.blocks.Convolution`?

File: monai/networks/blocks/segresnet_block.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        in_channels, out_channels, kernel_size=kernel_size, stride=stride, padding=kernel_size // 2, bias=bias,
+    )
+    return conv
+
+
+class ResBlock(nn.Module):
+    """
+    ResBlock emploies skip connection and two convolution blocks and is used
```
Comment (wyli) [inline monai/networks/blocks/segresnet_block.py:48]: typo: emploies  -> employs

**Comment by Nic-Ma** [comment]
Hi @wyli , 

We implemented this version based on the latest SegResNet implementation Andriy just sent us.
I can send you his original implementation if you like, he suggested to remove the VAE branch.

Thanks.

**Comment by wyli** [comment]
> Hi @wyli ,
> 
> We implemented this version based on the latest SegResNet implementation Andriy just sent us.
> I can send you his original implementation if you like, he suggested to remove the VAE branch.
> 
> Thanks.

Ok, I thought the goal here is to try our best to reimplement what the paper describes.
Plus implementing the generic VAE logic could be a very nice addition to MONAI
 cc @myron

**Comment by Nic-Ma** [comment]
Hi @wyli ,

@myron said the VAE part is not necessary anymore, need @myron to help provide more information here.
Thanks.

File: monai/networks/blocks/segresnet_block.py (hunk 1 above)
Comment (yiheng-wang-nv) [reply monai/networks/blocks/segresnet_block.py]: Thanks! I changed it.

File: monai/networks/blocks/segresnet_block.py (hunk 2 above)
Comment (yiheng-wang-nv) [reply monai/networks/blocks/segresnet_block.py]: thanks!

**Comment by yiheng-wang-nv** [comment]
> Hi @wyli ,
> 
> @myron said the VAE part is not necessary anymore, need @myron to help provide more information here.
> Thanks.

Hi @Nic-Ma @wyli , I've added VAE parts and updated the code.

**Comment by yiheng-wang-nv** [comment]
I see, let me refactor it

**Comment by yiheng-wang-nv** [comment]
> thanks, looks good to me, please add `SegResnetVAE` to `docs/source/networks.rst`, then I think this is ready

I see, I add it.
